src/controller.rs

```diff
@@ -60,49 +68,116 @@ async fn reconcile(obj: Arc<ServiceAlerts>, ctx: Arc<Context>) -> Result<Action,
+        let config_map_api: Api<ConfigMap> = Api::namespaced(ctx.client.clone(), &namespace);
+
+        let mut labels = BTreeMap::new();
+        labels.insert( "rules".to_string(), "prom-rule".to_string());
+
+        let mut data = BTreeMap::new();
+        data.insert("message".to_string(), "hello :)".to_string());
+
+        tracing::info!(
+            name=?&self.metadata.name,
+            namespace=?&self.metadata.namespace,
+            "building configmap"
+        );
+        let cm = ConfigMap {
+            metadata: ObjectMeta {
+                name: Some(name.clone()),
+                namespace: Some(namespace),
+                // This label is what allows prometheus to pick up the configMap
+                labels: Some(labels),
+
+                // finalizers: Some(vec![FINALIZER_NAME]),
+                owner_references: Some(vec![self.controller_owner_ref(&()).unwrap()]),
+                ..ObjectMeta::default()
+            },
+            data: Some(data),
+            // data: Some(contents),
+            ..Default::default()
+        };
+
+        tracing::info!(
+            name=?&self.metadata.name,
+            namespace=?&self.metadata.namespace,
+            "patching configmap"
+        );
+        config_map_api
+            .patch(
+                &name,
+                &PatchParams::apply(FINALIZER_NAME),
+                &Patch::Apply(&cm),
+            )
+            .await
+            .map_err(Error::ConfigMapCreationFailed).unwrap();
+
+
+
+        tracing::info!(
+            name=?&self.metadata.name,
+            namespace=?&self.metadata.namespace,
+            "building ServiceAlert status"
+        );
 
+        let requeue_duration: u64 = 5*60;
         let new_status = Patch::Apply(json!({
             "apiVersion": format!("{}/{}", API_GROUP, API_VERSION),
             "kind": KIND,
-            "status": ServiceAlertsStatus{
+            "status": ServiceAlertStatus{
                 last_reconciled_at: Some(Utc::now().format("%Y-%m-%dT%H:%M:%S").to_string()),
-                reconciliation_expires_at: Some((Utc::now() + chrono::Duration::seconds(30)).format("%Y-%m-%dT%H:%M:%S").to_string()),
+                reconciliation_expires_at: Some((Utc::now() + chrono::Duration::seconds(requeue_duration.clone() as i64)).format("%Y-%m-%dT%H:%M:%S").to_string()),
             }
         }));
+
+        tracing::info!(
+            name=?&self.metadata.name,
+            namespace=?&self.metadata.namespace,
+            "building ServiceAlert status"
+        );
         let ps = PatchParams::apply("cntrlr").force();
-        let _o = docs.patch_status(&name, &ps, &new_status).await?;
+
+        tracing::info!(
+            name=?&self.metadata.name,
+            namespace=?&self.metadata.namespace,
+            "getting ServiceAlert patch status"
+        );
+        let _o = service_alert_api.patch_status(&name, &ps, &new_status).await?;
```

Comment:
this is where it fails, it doesn't stop the configmap from being created but it's not ideal